shared/fs/row/container.js

```diff
@@ -10,37 +10,55 @@ type OwnProps = {
   path: Types.Path,
 }
 
-type DispatchProps = {
-  _onOpen: (type: Types.PathType, path: Types.Path) => void,
-  _openInFileUI: (path: Types.Path) => void,
-}
-
 const mapStateToProps = (state: TypedState, {path}: OwnProps) => {
-  const pathItem = state.fs.pathItems.get(path)
+  const pathItem = state.fs.pathItems.get(path) || Constants.makeUnknownPathItem()
   const _username = state.config.username || undefined
-  return {_username, path, type: pathItem ? pathItem.type : 'unknown'}
+  return {_username, path, pathItem}
 }
 
 const mapDispatchToProps = (dispatch: Dispatch) => ({
   _onOpen: (type: Types.PathType, path: Types.Path) => {
     if (type === 'folder') {
       dispatch(navigateAppend([{props: {path}, selected: 'folder'}]))
     } else {
-      dispatch(FSGen.createDownload({path}))
       console.log('Cannot view files yet. Requested file: ' + Types.pathToString(path))
     }
   },
   _openInFileUI: (path: Types.Path) => dispatch(FSGen.createOpenInFileUI({path: Types.pathToString(path)})),
+  _onAction: (
+    path: Types.Path,
+    pathItem: Types.PathItem,
+    itemStyles: Types.ItemStyles,
+    targetRect?: ?ClientRect
+  ) =>
+    dispatch(
+      navigateAppend([
+        {
+          props: {
+            path,
+            pathItem,
+            itemStyles,
```

Comment:
@chrisnojima this is still WIP but I had a question here:

Here I'm passing `path`, `pathItem`, and `itemStyles` through `routeProps` into `fs/row/popup-container.js`. The popup itself is connected, so it could actually get `pathItem` from the store, and compute `itemStyles` itself. But these are already available here, so I passed them in here to avoid recomputing them in the popup container.

On the other hand this means more stuff passed down and I roughly remember you mentioned in the chat refactor meeting that we should avoid passing down so many props and just make the child component connected. Could you help me understand what's the better approach in situation like this?

Reply:
don't pass it down. have those items be connected and if you want to share code just make a helper to compute that data. we don't want to couple components tightly, this should say 'show the menu for this path' and that's it.

Reply:
Got it; thanks!

Reply:
you removing all these other props?

Reply:
Oops I missed that. Will do. Thanks